src/silx/io/utils.py

```diff
@@ -1050,13 +1050,47 @@ def h5py_encode_value(value, encoding="utf-8", errors="surrogateescape"):
     """
     try:
         if numpy.isscalar(value):
-            return value.encode(encoding, errors=errors)
-        bytes_item = [s.encode(encoding, errors=errors) for s in value.flat]
+            return _encode_string(value, encoding, errors)
+        bytes_item = [_encode_string(s, encoding, errors=errors) for s in value.flat]
         return numpy.array(bytes_item, dtype=object).reshape(value.shape)
     except UnicodeEncodeError:
         return value
 
 
+def _decode_string(string, encoding, errors):
+    """
+    :param value: string
+    :param encoding str:
+    :param errors str:
+    """
+    try:
+        string = string.item()
+    except AttributeError:
+        pass  # not a numpy type
+    try:
+        return string.decode(encoding, errors=errors)
+    except AttributeError:
+        pass  # not bytes
+    return string
+
+
+def _encode_string(string, encoding, errors):
+    """
+    :param value: string
+    :param encoding str:
+    :param errors str:
+    """
+    try:
+        string = string.item()
+    except AttributeError:
+        pass  # not a numpy type
+    try:
+        return string.encode(encoding, errors=errors)
+    except AttributeError:
+        pass  # not bytes
+    return string
```

Comment:
What about this:
```suggestion
    if isinstance(string, bytes):
        return string
    return string.encode(encoding, errors=errors)
```

It should work since:
```python
In [70]: isinstance(numpy.bytes_(b"aaa"), bytes)
Out[70]: True
```